doc/usecases/current/membership_application.md

```diff
@@ -0,0 +1,42 @@
+# Use Case: Membership Application
+
+## Summary
+
+- **Scope:** Membership
+- **Level:** User Goal
+- **Actors:** User (Membership Applicant), Administrator (Chairman)
+- **Brief:** The user can apply for a membership (account)
+- **Assignee:** Thorben (Frontend), Lukas (Keycloak Installation)
+
+## Scenarios
+
+- **Precondition:**
+  The user has opened the app and is not logged in.
+- **Main success scenario:**
+  1. The user can click an register button which leads them to Keycloak to fill out:
+     - username
+     - first+last name
+     - email address
+  2. The user can look at public maps (read-only and without seeing addresses!)
+  3. The user can visit a 3-step form to apply for membership, with:
+     - user photo (for Nextcloud)
+     - telephone number (optional)
+     - website
+     - organisation
+     - experience (optional)
+     - photo of map
+     - location of map (not shown to public users)
```

Comment:
Why is this part of membership application? I understand that the map photo is usefull to verify the applicants intention of actually using the app for map planning, even though I find even that a little bit questionable (what is with users that only want to share expert knowledge and don't want to plan a map themselves?). But location should be at least optional. I wouldn't want to be forced to reveal geographic data when it isn't even guaranteed that the application will be approved.

Reply:
Good points! I fully agree that none of this data should be revealed (unless there is additional approval and only then it should be only within members) and this should be made very clear.

But: it is important that all of this is already in the membership application because we want high-quality meta-data for users and maps. So we let the user already enter the meta-data during onboarding, so that we can ensure high quality by simply rejecting applications if they have poor data, unacceptable photos, etc.

Btw. This also has implications on the database design: map_positions better should be an independent table that can exist before the map is created and might be used in several maps.

Btw2.: onboarding can already be something you use in your user study.

Reply:
> But nothing prevents the user to use other, possibly lower quality data for their actual maps. Even if they provide data for one meta-data rich map, additional maps could include data that is way below the bar initially set.

From my own experience I can say this will not often be the case. When people learn to do something properly and there is also an advantage of doing something properly (more people will find their map etc.), they also continue doing it properly.

So the first map is crucial because at that time many people simply don't know how to properly add metadata.

Also: for most people the first map will be the only map they create.

> Also, as I said, what is with users that do not want to create their own maps but instead use the collaborative features of PermaplanT? Why should they provide high-quality photos and location data for something they never use?

It hopefully will be an incentive to actually create a permaculture garden (and PermaplanT map) if they already have a nice place where they could do it. If they simply have a white board called "unnamed map" without photo their motivation will be very different.

Reply:
Btw. even if someone doesn't have a place, even adding a room (where room plants could be added) or some public place which could be beautified could be added as map. Actually, we should explicitly encourage people to always add one map, even if they don't have a garden.

Reply:
A quite obvious gamification of onboarding is if people can collect awards by completely filling out the forms.